### PR TITLE
Add bazel container image which includes build-essentials

### PR DESCRIPTION
**Pull request by saschagrunert**
Alternative solution to https://github.com/kubernetes/test-infra/pull/15613

### CODE REVIEW

File: images/README.md [hunk 1]
```diff
@@ -29,8 +29,9 @@ bit more on that [in the cloud-build docs][gcb_images].
 
 ## Currently used images
 
-Image                                     | used in/by
-:---:                                     | --
-[k8s-cloud-builder](./k8s-cloud-builder/) | The "main" image, [`anago`](../anago) runs with on cloud-build (submitted via [`gcbmgr`](../gcbmgr))
+| Image                                     | used in/by                                                                                           |
+| ----------------------------------------- | ---------------------------------------------------------------------------------------------------- |
+| [k8s-cloud-builder](./k8s-cloud-builder/) | The "main" image, [`anago`](../anago) runs with on cloud-build (submitted via [`gcbmgr`](../gcbmgr)) |
+| [prow](./prow)                            | The prow image used for CI testing                                                                   |
```
Comment (hoegaarden) [inline images/README.md]: Can we name the image `bazel`? I can easily imagine a world, where we (releng / k/release) maintain other images which we run on prow.

File: images/prow/cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+# See https://cloud.google.com/cloud-build/docs/build-config
+timeout: 1200s
+options:
+  substitution_option: ALLOW_LOOSE
```
Comment (hoegaarden) [inline images/prow/cloudbuild.yaml]: nit/question:
Why do we need that?
I'ts probably c&p, but still I'd like to understand why we want to explicitly enable this loose setting. Do you know why?

File: images/README.md (hunk 1 above)
Comment (saschagrunert) [reply images/README.md]: Sure, I rename it to `bazel`.

File: images/prow/cloudbuild.yaml (hunk 2 above)
Comment (saschagrunert) [reply images/prow/cloudbuild.yaml]: I have no idea I took this setting from the k8s-cloud-builder one. :man_shrugging:

**Comment by hoegaarden** [comment]
/lgtm

/hold
... for further discussion (also over [here](https://github.com/kubernetes/test-infra/pull/15613))

**Comment by justaugustus** [comment]
On second thought, let's keep the image pushes separate, so we don't obfuscate any individual image build failures.

**Comment by saschagrunert** [comment]
Okay, I renamed the image and created a parallel PR in test-infra: https://github.com/kubernetes/test-infra/pull/15835

**Comment by justaugustus** [comment]
/lgtm
/approve

Feel free to pull the hold once https://github.com/kubernetes/test-infra/pull/15835 merges.

**Comment by justaugustus** [comment]
https://github.com/kubernetes/test-infra/pull/15835 has merged.
/hold cancel

**Pull request merged**